### PR TITLE
408 relocation of server runtime files

### PR DESCRIPTION
**Pull request by naman108**

### CODE REVIEW

File: tests/test_controllers/MainConfig_test.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def test_env_var_list_values_with_comma(self):
 
     yaml_config = """
 System:
-    FTS_MAINLOOP_DELAY: 600
+    FTS_MAINLOOP_DELAY: 100
```
Comment (hickey) [inline tests/test_controllers/MainConfig_test.py:54]: @naman108 One of the reasons that this setting was set to an oddball value was to insure that the value from the YAML config was actually applied. The default value for `MainLoopDelay` is 100 and setting the same value in the YAML config makes it impossible to determine if the YAML configuration was applied correctly or not.

Comment (naman108) [reply tests/test_controllers/MainConfig_test.py:54]: oh, I misunderstood, it was failing when I executed the tests so I figured it was a typo, is it passing on your machine?

Comment (hickey) [reply tests/test_controllers/MainConfig_test.py:54]: Yes it was. I can pull the latest master and test again to make sure. Do you have the failure that you can post here?

Comment (hickey) [reply tests/test_controllers/MainConfig_test.py:54]: On `master` which still has a value of 600:

```
❯ python -m pytest -s tests/test_controllers/MainConfig_test.py
/Users/hickey/.pyenv/versions/fts/lib/python3.10/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
============================================================================== test session starts ==============================================================================
platform darwin -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0
rootdir: /Users/hickey/Projects/tak/fts
plugins: cov-4.0.0, pep8-1.0.6
collected 12 items

tests/test_controllers/MainConfig_test.py ............

============================================================================== 12 passed in 0.16s ===============================================================================
```

File: FreeTAKServer/controllers/configuration/MainConfig.py [hunk 2]
```diff
@@ -32,6 +34,10 @@ class MainConfig:
     except:
         _ip = "0.0.0.0"
 
+    # create the persistence path if it doesn't exist
+    if not os.path.exists(PERSISTENCE_PATH):
+        os.mkdir(PERSISTENCE_PATH)
```
Comment (hickey) [inline FreeTAKServer/controllers/configuration/MainConfig.py:39]: We need to be checking the result of `mkdir` and if it fails, then catch the exception. This will allow us to put a friendlier error message in place. This will also help inexperienced users and hopefully lessen the amount of support that we need to do.

File: FreeTAKServer/controllers/configuration/MainConfig.py [hunk 3]
```diff
@@ -2,16 +2,18 @@
...
 
 # the version information of the server (recommended to leave as default)
-FTS_VERSION = 'FreeTAKServer-1.9.9.6 Public'
+FTS_VERSION = 'FreeTAKServer-1.9.9.7 Public'
 API_VERSION = '1.9.5'
 # TODO Need to find a better way to determine python version at runtime
 PYTHON_VERSION = 'python3.8'
 USERPATH = '/usr/local/lib/'
+PERSISTENCE_PATH = '/opt/fts/'
 MAINPATH = fr'{USERPATH}{PYTHON_VERSION}/dist-packages/FreeTAKServer'
```
Comment (hickey) [inline FreeTAKServer/controllers/configuration/MainConfig.py:17]: I think that `MAINPATH` pretty much goes away at this point. Which is fine by me. I had started a branch to detect what version of Python was in use to auto configure the `MAINPATH`, but with this PR it should not be needed.

File: FreeTAKServer/controllers/configuration/MainConfig.py (hunk 2 above)
Comment (naman108) [reply FreeTAKServer/controllers/configuration/MainConfig.py]: also addressed with the most recent commit but could you validate the error message in terms of user friendliness

File: FreeTAKServer/controllers/configuration/MainConfig.py [hunk 4]
```diff
@@ -32,6 +31,14 @@ class MainConfig:
     except:
         _ip = "0.0.0.0"
 
+    # create the persistence path if it doesn't exist
+    if not os.path.exists(PERSISTENCE_PATH):
```
Comment (dlc-ariel) [inline FreeTAKServer/controllers/configuration/MainConfig.py:35]: We should also check that the path can be written to, using `os.access(PERSISTENCE_PATH, os.W_OK)`

File: FreeTAKServer/controllers/configuration/MainConfig.py [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
 
 # the version information of the server (recommended to leave as default)
-FTS_VERSION = 'FreeTAKServer-1.9.9.6 Public'
+FTS_VERSION = 'FreeTAKServer-1.9.9.7 Public'
 API_VERSION = '1.9.5'
-# TODO Need to find a better way to determine python version at runtime
-PYTHON_VERSION = 'python3.8'
-USERPATH = '/usr/local/lib/'
-MAINPATH = fr'{USERPATH}{PYTHON_VERSION}/dist-packages/FreeTAKServer'
+PERSISTENCE_PATH = r'/opt/fts/'
```
Comment (dlc-ariel) [inline FreeTAKServer/controllers/configuration/MainConfig.py:14]: Thought, what if we make this value solely an environment variable rather then a yaml config variable? Since the config overwrite does not work correctly, grabbing this from the environment beforehand will solve the issues we are having where accessing any attribute that relies on this value.

Otherwise we will need to use the template system @naman108 mentioned

File: tests/test_controllers/MainConfig_test.py [hunk 6]
```diff
@@ -114,17 +114,17 @@ def test_yaml_config(self):
...
 
         assert(config.SSLCoTServicePort== 10000)
         assert(config.SaveCoTToDB == True)
-        assert(config.MainPath == '/tmp/fts')
+        assert(config.persistencePath == '/tmp/fts')
 
     def test_get_config_as_attribute(self):
         config = MainConfig.instance()
-        assert(config.MainPath == fr'{USERPATH}{PYTHON_VERSION}/dist-packages/FreeTAKServer')
+        assert(config.persistencePath == fr'/opt/fts/')
```
Comment (brothercorvo) [inline tests/test_controllers/MainConfig_test.py:127]: > __[Pylint]__ Unnecessary parens after 'assert' keyword ([view](https://sider.review/gh/repos/238457011/pulls/420#issue-8341084))
>
> <details>
>
> | Rule | Severity | Message-id | Module | Obj |
> | ---- | -------- | ---------- | ------ | --- |
> | `superfluous-parens` | `convention` | C0325 | MainConfig_test | - |
>
> You can [close this issue](https://sider.review/gh/repos/238457011/pulls/420#issue-8341084) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

we have move all the tests out, for now so that should not be part of the commit

**Comment by brothercorvo** [comment]
@naman108 what's is missing to finish this?

**Comment by brothercorvo** [comment]
@naman108 what's the status of this?

**Comment by brothercorvo** [comment]
We need to merge this into Master in order to prepare for Alpha release @hickey
